src/pages/signup/Signup.js

```diff
@@ -1,7 +1,67 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 import './Signup.scss';
 
 const Signup = () => {
+  const [userData, setUserData] = useState({
+    email: '',
+    password: '',
+    confirmPassword: '',
+    userName: '',
+    firstNumber: '010',
+    secondNumber: '',
+    lastNumber: '',
+  });
+
+  const [emailError, setEmailError] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [numberError, setNumberError] = useState(false);
+
+  const handleChange = event => {
+    const { name, value } = event.target;
+    setUserData(prevData => {
+      return { ...prevData, [name]: value };
+    });
+  };
+  const matchPassword = userData.password === userData.confirmPassword;
+  const validPassword = /^[A-Za-z0-9]{8,20}$/;
+  const validName = userData.userName.length > 0;
+  const validNumber =
+    (userData.secondNumber.length && userData.lastNumber.length) === 4;
+
+  const handleClick = () => {
+    if (!matchPassword) {
+      return;
+    } else if (!validName) {
+      setNameError(true);
+    } else if (!validNumber) {
+      setNumberError(true);
+    } else {
+      fetch('http://10.58.52.137:3000/users/signup', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json;charset=utf-8' },
+        body: JSON.stringify({
+          email: userData.email,
+          password: userData.password,
+          mobile: `${userData.firstNumber}-${userData.secondNumber}-${userData.lastNumber}`,
+          username: userData.userName,
+        }),
+      })
+        .then(res => {
+          if (res.ok) {
+            alert('회원가입을 축하합니다.');
+            navigate('/Login');
+          }
```

Comment:
```suggestion
          if (!res.ok) return;
          alert('회원가입을 축하합니다.');
          navigate('/Login');
```

이런 식으로도 작성할 수 있겠네요.
이러한 방식을 `early-return 패턴`이라고 합니다!